lib/core/logconf.c

```diff
@@ -487,6 +496,13 @@ char *h2o_log_request(h2o_logconf_t *logconf, h2o_req_t *req, size_t *len, char
             RESERVE(sizeof(H2O_UINT16_LONGEST_STR) - 1);
             pos = append_port(pos, req->conn->callbacks->get_peername, req->conn, nullexpr);
             break;
+        case ELEMENT_TYPE_ENV_VAR:  /* %{..}e */  {
+            h2o_iovec_t *env_var = h2o_req_getenv(req, element->data.name.base, element->data.name.len, 0);
+            if (env_var == NULL || env_var->len < 1)
```

Comment:
Shouldn't we emit an zero-length string if `len` is zero? Some applications may use the existence of the environment variable as a sign of something, and in such case it is desirable to be able to distinguish between NULL and a zero-length string.